apps/discovery_api/test/integration/discovery_api/auth/token_handler_test.exs

```diff
@@ -0,0 +1,231 @@
+      claims = claims()
+      token = "also_ignored"
+      options = []
+
+      {:ok, _} = TokenHandler.on_revoke(
+        claims,
+        token,
+        []
+      )
+
+      assert {:error, _} = TokenHandler.on_verify(
+        claims,
+        token,
+        options
+      )
+    end
+
+    test "given a revoked token/claim, and an attempt to store, it returns an error" do
+      claims = claims()
+      token = "also_ignored"
+      options = [store_token: true]
+
+      {:ok, _} = TokenHandler.on_revoke(
+        claims,
+        token,
+        []
+      )
+
+      assert {:error, _} = TokenHandler.on_verify(
+        claims,
+        token,
+        options
+      )
+
+      assert {:error, _} = TokenHandler.on_verify(
+        claims,
+        token,
+        []
+      )
+    end
+
+    test "given a revoked token/claim, and the token somehow getting back in the database, it does not return an error" do
+      resource = :ignored
+      claims = claims()
+      token = "also_ignored"
+      options = []
+
+      {:ok, _} = TokenHandler.on_revoke(
+        claims,
+        token,
+        []
+      )
+
+      {:ok, _} = TokenHandler.after_encode_and_sign(
+        resource,
+        claims,
+        token,
+        []
+      )
+
+      assert {:ok, _} = TokenHandler.on_verify(
+        claims,
+        token,
+        options
+      )
+    end
+
+    test "given a revoked token/claim, and option to store, it returns an error" do
+      claims = claims()
+      token = "also_ignored"
+      options = [store_token: true]
+
+      {:ok, _} = TokenHandler.on_revoke(
+        claims,
+        token,
+        []
+      )
+
+      assert {:error, _} = TokenHandler.on_verify(
+        claims,
+        token,
+        options
+      )
+    end
+  end
+
+  describe "on_revoke/3" do
+    test "given a complete token, it removes it from the database and puts a revoked marker in place" do
+      claims = claims()
+      token = "also_ignored"
+      options = []
+
+      assert {:ok, ^claims} = TokenHandler.on_revoke(
+        claims,
+        token,
+        options
+      )
+
+      claims_primary_key = TokenHandler.to_storable_claims(claims)
+      revoked_primary_key = TokenHandler.to_revoked_claims(claims)
+
+      assert nil == Guardian.DB.Token.find_by_claims(claims_primary_key)
+      assert nil != Guardian.DB.Token.find_by_claims(revoked_primary_key)
+    end
+
+    test "given an incomplete token, it returns an error tuple" do
+      claims_without_aud = claims()
+      |> Map.delete("aud")
+      token = "also_ignored"
+      options = []
+
+      assert {:error, _} = TokenHandler.on_revoke(
+        claims_without_aud,
+        token,
+        options
+      )
+    end
+  end
+
+  defp claims(overrides \\ %{}) do
+    demo_issuer = "https://smartcolumbusos-demo.auth0.com/"
+    demo_client_id = "sfe5fZzFXsv5gIRXz8V3zkR7iaZBMvL0"
+    darsh_subject = "auth0|5e3066daf048aa0e71bdd77e"
```

Comment:
are any of these legit?  If so, and they do not need to be please obfuscate.

Reply:
Yeah, and they can be masked. Everything but the keyid, honestly